Marginal particle detection probabilities (unitary hack)

Bread-Pitt64 opened the pull request:
This PR introduces the get_marginal_fock_probabilities method for the Gaussianstate, Fockstate and Purefickstate classes and adds a test file to tests/_simulators

Kolarovszki reviewed (changes requested):
Thank you for your PR!

Bread-Pitt64 commented on piquasso/_simulators/fock/general/state.py:
```diff
@@ -201,6 +201,10 @@ def reduced(self, modes: Tuple[int, ...]) -> "FockState":
 
         return reduced_state
 
+    def get_marginal_fock_probabilities(self, modes) -> np.ndarray:
```
You're completely right, i'll add that right away

Bread-Pitt64 commented on tests/test_get_marginal_fock_probabilities.py:
```diff
@@ -0,0 +1,101 @@
+#
```
I just realized i pushed an earlier version of the tests, i made some that gaussian states as well, when i can ill push that version, sorry for the inconvinience.

Bread-Pitt64 replied:
Oh of course not, i forgot to remove the earlier version, in the next patch ill take care of that

Bread-Pitt64 commented:
In this version of the test file most test are used for all the state classes, if you still want me to separate them in the 3 folders just say

Kolarovszki commented on tests/_simulators/test_get_marginal_fock_probabilities.py:
```diff
@@ -0,0 +1,282 @@
+        pq.FockSimulator,
+    ),
+)
+def test_get_marginal_fock_probabilities_with_displacement(SimulatorClass):
```
I don't understand this, as the asserts are very similar as in other tests.

Kolarovszki commented on tests/_simulators/test_get_marginal_fock_probabilities.py:
```diff
@@ -0,0 +1,282 @@
+        pq.FockSimulator,
+    ),
+)
+def test_get_marginal_fock_probabilities_mode_ordering(SimulatorClass):
```
I don't understand this, as it doesn't check if the two probability distributions are equivalent.

Bread-Pitt64 commented:
Ok now i think it's everything.
I left only the test you said were of interest, made the change you sugested to the single photon test, added an example with explicit numbers (for modes (0,1) and (1,2)), to do this i coppied the is_proportional function, deleted that other file and formnatted everything with black.
If you need anything else just say :).

Kolarovszki commented:
> Ok now i think it's everything. I left only the test you said were of interest, made the change you sugested to the single photon test, added an example with explicit numbers (for modes (0,1) and (1,2)), to do this i coppied the is_proportional function, deleted that other file and formnatted everything with black. If you need anything else just say :).

Thank you so much! This patch looks good now IMO, however, the pipeline is failing (maybe flake8 issue?)

Kolarovszki commented:
I think this PR is good at this point, but since the pipeline is failing, I cannot approve it, and in the meanwhile another PR got accepted (https://github.com/Budapest-Quantum-Computing-Group/piquasso/pull/453). Therefore I need to close this PR now. Sorry for the inconvenience.
